### PR TITLE
gcc: modernize

### PR DESCRIPTION
**Pull request by dawidd6**
This is an automated pull request to build a new bottle for linuxbrew-core
based on the existing bottle block from homebrew-core.

### CODE REVIEW

File: Formula/gcc.rb [hunk 1]
```diff
@@ -42,15 +32,17 @@ def arch
   end
 
   unless OS.mac?
-    depends_on "zlib"
-    depends_on "binutils" if build.with? "glibc"
-    depends_on "glibc" => (Formula["glibc"].installed? || OS::Linux::Glibc.system_version < Formula["glibc"].version) ? :recommended : :optional
```
Comment (sjackman) [inline Formula/gcc.rb]: Please leave the `--without-glibc` option. It's non-standard to have options in core, but it's sometimes useful to bootstrap on older systems.

File: Formula/gcc.rb [hunk 2]
```diff
@@ -108,7 +112,7 @@ def install
       ln_s Utils.popen_read(ENV.cc, "-print-file-name=libstdc++.so.6").strip, lib
       ln_s Utils.popen_read(ENV.cc, "-print-file-name=libgcc_s.so.1").strip, lib
 
-      if build.with? "glibc"
+      if Formula["glibc"].any_version_installed?
```
Comment (sjackman) [inline Formula/gcc.rb]: Ditto.

**Comment by iMichka** [comment]
> Please leave the --without-glibc option. Otherwise looks good. Thanks!

Could we have another gcc formula without glibc maybe?

**Comment by sjackman** [comment]
Thanks, Dawid!

**Comment by sjackman** [comment]
>> Please leave the --without-glibc option. Otherwise looks good. Thanks!

> Could we have another gcc formula without glibc maybe?

That's an option. I'd leave it as is for now, until we again run into a specific use case for building `gcc` without `glibc`. Kind of related but unrelated, but a second `glibc` formula that is `cellar :any` could be really useful for bootstrapping on older systems. It would be used only to run `gcc` to compile `glibc`, and then could be removed.

**Comment by sjackman** [comment]
It would be `cellar :any` only be decree. It would operate in a somewhat degraded fashion, because it's not actually `cellar :any`, but good enough to compile `gcc`.

**Comment by iMichka** [comment]
Yes, let's leave it as-is for now.

Just FYI, `gcc --without-glibc` is 723 installs since Jan 2019, vs 77,191 installs for plain gcc.

**Pull request closed without merging**